### ISSUE TITLE
Unable to modify Prefix + <C-l> binding

### ISSUE THREAD

**Issue by Tycholiz** [tmux/tmux#4160]
### System info
- tmux v3.5
- MacOS Sequoia v15.0

### Issue description
I am experiencing an issue with tmux where the key binding PREFIX + <C-l> does not function as expected. Instead of switching to the next window, it clears the terminal screen, which is the default behavior of zsh.

### Steps to reproduce:
1. Start a new tmux session.
2. Attempt to use the key combination `PREFIX + <C-l>` to switch windows.
3. Observe that the terminal screen clears instead of switching windows.

#### Configuration:
I have verified that other key bindings (like `PREFIX + <C-k>`, `PREFIX + <C-h>`, and `PREFIX + <C-j>`) work correctly. I have also attempted to override the `<C-l>` binding in my `.tmux.conf` file with the following lines:

```
unbind-key -T prefix C-l
bind-key -T prefix C-l next-window
```

Despite this, the key binding does not work as intended.

Current key bindings related to `<C-l>`:
Running `tmux list-keys` yields the following relevant entries:

```
bind-key    -T copy-mode-vi     C-l                    select-pane -R
bind-key    -T prefix                    C-l                    send-keys C-l
```

As we can see by the second entry, Tmux is configured to just pass the <C-l> combination to the terminal (zsh).

To compare, here are the current key bindings related to `<C-h>`, as expected:
```
bind-key    -T copy-mode-vi C-h                    select-pane -L
bind-key -r -T prefix              C-h                    previous-window
```

### Other
- I've observed this same issue on the native Terminal app, so it doesn't appear to be originating with the terminal itself.

**Comment by nicm** [comment]
Logs?

**Comment by Tycholiz** [comment]
is there any part of the logs that I can paste for you?

the client logs are 286 lines long, but the server logs are almost 10,000. 

Are you interested in the tail 100 lines or so?

**Comment by nicm** [comment]
Attach the entire server log to this issue please.

**Comment by Tycholiz** [comment]
[tmux-server-4811.log](https://github.com/user-attachments/files/17264334/tmux-server-4811.log)

**Comment by nicm** [comment]
I do not see you pressing `C-l` in this log, only `C-b`, which appears to work?

**Comment by nicm** [comment]
Sorry, I missed it, I see it now.

**Comment by nicm** [comment]
Your key binding for `C-l` is `send-keys C-l` and that is what tmux is doing:
~~~~
1728080483.291280 /dev/ttys000: keys are 1 (\f)
1728080483.291285 /dev/ttys000: next key is 1 (\f) (expired=0)
1728080483.291290 /dev/ttys000: complete key \f 0x20000000006c
...
1728080483.291337 key table prefix (pane %0)
1728080483.291344 found in key table prefix
1728080483.291353 cmdq_get_command: [send-keys/0x1586101e0] group 3759
1728080483.291357 cmdq_insert_after </dev/ttys000>: [send-keys/0x1586101e0] after [server_client_key_callback/0x1586149d0]
1728080483.291360 unref client 0x15900c000 (3 references)
1728080483.291364 cmdq_next </dev/ttys000>: [send-keys/0x1586101e0] (0), flags 0
1728080483.291373 message: /dev/ttys000 key C-l: send-keys C-l
~~~~

**Comment by nicm** [comment]
If you look around line 57184 of the log, a client is attaching and setting the `C-l` binding to `send-keys C-l`. Did you do this? If not, you will need to track down whatever did.

**Comment by Tycholiz** [comment]
Well as far as I know my main session was the only client.

By any means, I don't have this binding set in my config, so I'm not sure where it could possibly be coming from

Here are my tmux conf files:
[tmux.conf.txt](https://github.com/user-attachments/files/17264397/tmux.conf.txt)
[tmux.conf.local.txt](https://github.com/user-attachments/files/17264398/tmux.conf.local.txt)

**Comment by nicm** [comment]
I suggest you start without a config and add things from there until you find out what is doing it.

**Comment by Tycholiz** [comment]
I seem to be finding myself dealing with some strange behaviour.

I removed **all** content from the `.tmux.conf` file and sourced the file, and noted that the behaviour persisted. Then, I added `bind-key -T prefix C-l next-window` to the config file and sourced it again, and got the expected behaviour! Oddly however, when I remove the line again and source the config, I still get the expected behaviour.

I have a case here where 2 identical .tmux.conf files result in different behaviour. This doesn't seem like it should be possible

**Comment by nicm** [comment]
Are you sure this isn't being changed from a shell profile or there isn't a stray script running that is doing this?

**Comment by nicm** [comment]
> I removed all content from the .tmux.conf file and sourced the file

You need to restart tmux, sourcing an empty `.tmux.conf` will not change existing configuration.

**Comment by Tycholiz** [comment]
After restarting my computer, the issue doesn't seem to be persisting anymore, but aside from what I described above (with the empty config), I was still observing the behaviour I described (2 identical .tmux.conf files result in different behaviour), having had put content in the file before sourcing it.

It seems like this is a completely separate issue than my original one, but it seems my original one was being caused by a plugin (`christoomey/vim-tmux-navigator`).

Not really sure what was happening there. Maybe it will resurface. All good for now though, thanks so much for your help
